<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.262">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Evgenii O. Tretiakov">
<meta name="dcterms.date" content="2023-07-07">

<title>Differential expression analysis of Hypothalamus astrocytes dataset from Romanov RA et al 2020</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="04-de_test-dendrogram_files/libs/clipboard/clipboard.min.js"></script>
<script src="04-de_test-dendrogram_files/libs/quarto-html/quarto.js"></script>
<script src="04-de_test-dendrogram_files/libs/quarto-html/popper.min.js"></script>
<script src="04-de_test-dendrogram_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="04-de_test-dendrogram_files/libs/quarto-html/anchor.min.js"></script>
<link href="04-de_test-dendrogram_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="04-de_test-dendrogram_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="04-de_test-dendrogram_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="04-de_test-dendrogram_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="04-de_test-dendrogram_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">

<link href="04-de_test-dendrogram_files/libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="04-de_test-dendrogram_files/libs/pagedtable-1.1/js/pagedtable.js"></script>


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#load-data-and-setup-parameters" id="toc-load-data-and-setup-parameters" class="nav-link active" data-scroll-target="#load-data-and-setup-parameters">Load data and setup parameters</a>
  <ul class="collapse">
  <li><a href="#set-paths" id="toc-set-paths" class="nav-link" data-scroll-target="#set-paths">Set paths</a></li>
  <li><a href="#load-helper-functions-and-gene-sets" id="toc-load-helper-functions-and-gene-sets" class="nav-link" data-scroll-target="#load-helper-functions-and-gene-sets">Load helper functions and gene-sets</a></li>
  <li><a href="#set-fixed-variables" id="toc-set-fixed-variables" class="nav-link" data-scroll-target="#set-fixed-variables">Set fixed variables</a></li>
  </ul></li>
  <li><a href="#load-predicted-astrocytes-data-and-subset-from-romanov-ra-et-al-2020" id="toc-load-predicted-astrocytes-data-and-subset-from-romanov-ra-et-al-2020" class="nav-link" data-scroll-target="#load-predicted-astrocytes-data-and-subset-from-romanov-ra-et-al-2020">Load predicted astrocytes data and subset from Romanov RA et al 2020</a>
  <ul class="collapse">
  <li><a href="#convert-adata-object-to-r-anndatar6-object." id="toc-convert-adata-object-to-r-anndatar6-object." class="nav-link" data-scroll-target="#convert-adata-object-to-r-anndatar6-object.">Convert adata object to R AnnDataR6 object.</a></li>
  </ul></li>
  <li><a href="#session-information" id="toc-session-information" class="nav-link" data-scroll-target="#session-information">Session information</a></li>
  </ul>
</nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Differential expression analysis of Hypothalamus astrocytes dataset from Romanov RA et al 2020</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Evgenii O. Tretiakov </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">July 7, 2023</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="load-data-and-setup-parameters" class="level2">
<h2 class="anchored" data-anchor-id="load-data-and-setup-parameters">Load data and setup parameters</h2>
<div class="cell" data-layout-align="center">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load tidyverse infrastructure packages</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>({</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(future)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(here)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(tidyverse)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(magrittr)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(stringr)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(skimr)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(RColorBrewer)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(viridis)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Load packages for scRNA-seq analysis and visualisation</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>({</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(ggplot2)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(cowplot)</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(patchwork)</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(ggstatsplot)</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(anndata)</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(sceasy)</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(Seurat)</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(SeuratDisk)</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(SeuratWrappers)</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(scCustomize)</span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a>sc <span class="ot">&lt;-</span> <span class="fu">import</span>(<span class="st">"scanpy"</span>, <span class="at">convert =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="set-paths" class="level3">
<h3 class="anchored" data-anchor-id="set-paths">Set paths</h3>
<div class="cell" data-layout-align="center">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>src_dir <span class="ot">&lt;-</span> <span class="fu">here</span>(<span class="st">"code"</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>data_dir <span class="ot">&lt;-</span> <span class="fu">here</span>(<span class="st">"../data"</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>output_dir <span class="ot">&lt;-</span> <span class="fu">here</span>(<span class="st">"output"</span>)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>plots_dir <span class="ot">&lt;-</span> <span class="fu">here</span>(output_dir, <span class="st">"figures"</span>)</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>tables_dir <span class="ot">&lt;-</span> <span class="fu">here</span>(output_dir, <span class="st">"tables"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="load-helper-functions-and-gene-sets" class="level3">
<h3 class="anchored" data-anchor-id="load-helper-functions-and-gene-sets">Load helper functions and gene-sets</h3>
<div class="cell" data-layout-align="center">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">here</span>(src_dir, <span class="st">"genes.R"</span>))</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">here</span>(src_dir, <span class="st">"functions.R"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="set-fixed-variables" class="level3">
<h3 class="anchored" data-anchor-id="set-fixed-variables">Set fixed variables</h3>
<div class="cell" data-layout-align="center">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># set seed</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>reseed <span class="ot">&lt;-</span> <span class="dv">42</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="at">seed =</span> reseed)</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Parameters for parallel execution</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>n_cores <span class="ot">&lt;-</span> <span class="dv">32</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="fu">plan</span>(<span class="st">"multisession"</span>, <span class="at">workers =</span> n_cores)</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">future.globals.maxSize =</span> <span class="dv">100000</span> <span class="sc">*</span> <span class="dv">1024</span><span class="sc">^</span><span class="dv">2</span>,</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">future.rng.onMisuse =</span> <span class="st">"ignore"</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a><span class="fu">plan</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>multisession:
- args: function (..., workers = 32, envir = parent.frame())
- tweaked: TRUE
- call: plan("multisession", workers = n_cores)</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ggplot2 theme</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="fu">theme_set</span>(ggmin<span class="sc">::</span><span class="fu">theme_powerpoint</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-layout-align="center">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>bioproject <span class="ot">&lt;-</span> <span class="st">"PRJNA548917"</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>project <span class="ot">&lt;-</span> <span class="st">"romanov2020_Hypoth-dev"</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>cb_fpr <span class="ot">&lt;-</span> <span class="fl">0.001</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>low_cutoff_gene <span class="ot">&lt;-</span> <span class="dv">500</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>high_cutoff_gene <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>high_cutoff_gene <span class="ot">&lt;-</span> <span class="dv">6000</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>low_cutoff_umis <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>low_cutoff_umis <span class="ot">&lt;-</span> <span class="sc">-</span><span class="cn">Inf</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>high_cutoff_umis <span class="ot">&lt;-</span> <span class="dv">20000</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>high_cutoff_pc_mt <span class="ot">&lt;-</span> <span class="dv">15</span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>high_cutoff_pc_ribo <span class="ot">&lt;-</span> <span class="dv">15</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>high_cutoff_pc_hb <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>high_cutoff_doublet_score <span class="ot">&lt;-</span> <span class="fl">0.33</span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>high_cutoff_complexity <span class="ot">&lt;-</span> <span class="fl">0.85</span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>connectivity_model <span class="ot">&lt;-</span> <span class="st">"min_tree"</span></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>k <span class="ot">&lt;-</span> <span class="dv">10</span></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>metric <span class="ot">&lt;-</span> <span class="st">"euclidean"</span></span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>signature <span class="ot">&lt;-</span> <span class="dv">100</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="load-predicted-astrocytes-data-and-subset-from-romanov-ra-et-al-2020" class="level2">
<h2 class="anchored" data-anchor-id="load-predicted-astrocytes-data-and-subset-from-romanov-ra-et-al-2020">Load predicted astrocytes data and subset from Romanov RA et al 2020</h2>
<div class="cell" data-layout-align="center">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>anndata <span class="ot">&lt;-</span> sc<span class="sc">$</span><span class="fu">read</span>(<span class="fu">here</span>(</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>  data_dir,</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sprintf</span>(<span class="st">"resolved_subregions_by_microclusters/best_xgboost-subregional_%s-astrocytes_dataset-msp_%s-metric_%s-k_%s-sign_%s-amb_%s.h5ad"</span>, bioproject, connectivity_model, metric, k, signature, cb_fpr)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="convert-adata-object-to-r-anndatar6-object." class="level3">
<h3 class="anchored" data-anchor-id="convert-adata-object-to-r-anndatar6-object.">Convert adata object to R AnnDataR6 object.</h3>
<div class="cell" data-layout-align="center">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>adata <span class="ot">&lt;-</span> <span class="fu">py_to_r</span>(anndata)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(adata)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "AnnDataR6" "R6"       </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(adata<span class="sc">$</span>X)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "matrix" "array" </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>adata</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>AnnData object with n_obs × n_vars = 3042 × 21989
    obs: 'nCount_RAW', 'nFeature_RAW', 'nCount_RNA', 'nFeature_RNA', 'orig.ident', 'nFeature_Diff', 'nCount_Diff', 'percent_mito', 'percent_ribo', 'percent_mito_ribo', 'percent_hb', 'log10GenesPerUMI', 'cell_name', 'barcode', 'latent_RT_efficiency', 'latent_cell_probability', 'latent_scale', 'doublet_score', 'predicted_doublets', 'QC', 'var_regex', 'RNA_snn_res.0.5', 'RNA_snn_res.0.730162924108181', 'RNA_snn_res.1.22140339759076', 'RNA_snn_res.3.000001', 'seurat_clusters', 'k_tree', 'comb_clstr1', 'S.Score', 'G2M.Score', 'Phase', 'nCount_SCT', 'nFeature_SCT', 'SCT_snn_res.1', 'SCT_snn_res.1.10049753133357', 'SCT_snn_res.1.22022353686838', 'SCT_snn_res.1.36528036285893', 'SCT_snn_res.1.54465801813792', 'SCT_snn_res.1.77217142500043', 'SCT_snn_res.2.0701989287267', 'SCT_snn_res.2.47753434006845', 'SCT_snn_res.3.06781889081486', 'SCT_snn_res.4.00000100000001', 'bioproject', 'project', 'model', 'tech', 'region', 'sex', 'stage', 'libname', 'expbtch', 'condit', 'ora_celltype', 'train', 'test', 'Regulon(Ahr)', 'Regulon(Atf3)', 'Regulon(Brca1)', 'Regulon(Dlx5)', 'Regulon(E2f1)', 'Regulon(E2f2)', 'Regulon(E2f7)', 'Regulon(Ebf1)', 'Regulon(Egr2)', 'Regulon(Elf4)', 'Regulon(Fli1)', 'Regulon(Fos)', 'Regulon(Fosb)', 'Regulon(Fosl1)', 'Regulon(Foxo1)', 'Regulon(Gata1)', 'Regulon(Gata2)', 'Regulon(Gata3)', 'Regulon(Gata4)', 'Regulon(Hes7)', 'Regulon(Hnf4g)', 'Regulon(Ikzf1)', 'Regulon(Jun)', 'Regulon(Junb)', 'Regulon(Jund)', 'Regulon(Klf2)', 'Regulon(Mafk)', 'Regulon(Mef2c)', 'Regulon(Mxi1)', 'Regulon(Nfe2l3)', 'Regulon(Nfil3)', 'Regulon(Nfya)', 'Regulon(Nkx6-2)', 'Regulon(Nr1d1)', 'Regulon(Pax5)', 'Regulon(Prdm1)', 'Regulon(Rarg)', 'Regulon(Rfx4)', 'Regulon(Runx1)', 'Regulon(Runx2)', 'Regulon(Sox9)', 'Regulon(Sp2)', 'Regulon(Sp5)', 'Regulon(Stat3)', 'Regulon(Tal1)', 'Regulon(Tbx19)', 'Regulon(Tbx21)', 'Regulon(Tcf4)', 'Regulon(Tead1)', 'Regulon(Zfp109)', 'Regulon(Zfp319)', 'Regulon(Zic3)', 'predict_logit_subregion', 'predict_xgboost_subregion'
    var: 'vst.mean', 'vst.variance', 'vst.variance.expected', 'vst.variance.standardized', 'vst.variable', 'Regulon(Ahr)', 'Regulon(Atf3)', 'Regulon(Brca1)', 'Regulon(Dlx5)', 'Regulon(E2f1)', 'Regulon(E2f2)', 'Regulon(E2f7)', 'Regulon(Ebf1)', 'Regulon(Egr2)', 'Regulon(Elf4)', 'Regulon(Fli1)', 'Regulon(Fos)', 'Regulon(Fosb)', 'Regulon(Fosl1)', 'Regulon(Foxo1)', 'Regulon(Gata1)', 'Regulon(Gata2)', 'Regulon(Gata3)', 'Regulon(Gata4)', 'Regulon(Hes7)', 'Regulon(Hnf4g)', 'Regulon(Ikzf1)', 'Regulon(Jun)', 'Regulon(Junb)', 'Regulon(Jund)', 'Regulon(Klf2)', 'Regulon(Mafk)', 'Regulon(Mef2c)', 'Regulon(Mxi1)', 'Regulon(Nfe2l3)', 'Regulon(Nfil3)', 'Regulon(Nfya)', 'Regulon(Nkx6-2)', 'Regulon(Nr1d1)', 'Regulon(Pax5)', 'Regulon(Prdm1)', 'Regulon(Rarg)', 'Regulon(Rfx4)', 'Regulon(Runx1)', 'Regulon(Runx2)', 'Regulon(Sox9)', 'Regulon(Sp2)', 'Regulon(Sp5)', 'Regulon(Stat3)', 'Regulon(Tal1)', 'Regulon(Tbx19)', 'Regulon(Tbx21)', 'Regulon(Tcf4)', 'Regulon(Tead1)', 'Regulon(Zfp109)', 'Regulon(Zfp319)', 'Regulon(Zic3)'
    uns: 'aucell', 'k_tree_colors', 'log1p', 'name', 'ora_celltype_colors', 'predict_logit_subregion_colors'
    obsm: 'X_aucell', 'X_pacmap', 'X_pca', 'X_umap', 'X_umap_logit', 'X_umap_subfunct', 'ora_estimate', 'ora_pvals'</code></pre>
</div>
</div>
<div class="cell" data-layout-align="center">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>srt_path <span class="ot">&lt;-</span> <span class="fu">here</span>(</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>  data_dir,</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sprintf</span>(<span class="st">"resolved_subregions_by_microclusters/best_xgboost-subregional_%s-astrocytes_dataset-msp_%s-metric_%s-k_%s-sign_%s-amb_%s.h5Seurat"</span>, bioproject, connectivity_model, metric, k, signature, cb_fpr)</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>regsrt_path <span class="ot">&lt;-</span> <span class="fu">here</span>(</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>  data_dir,</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sprintf</span>(<span class="st">"resolved_subregions_by_microclusters/%s_regulons-DER-astrocytes_dataset-msp_%s-metric_%s-k_%s-sign_%s-amb_%s.h5Seurat"</span>, bioproject, connectivity_model, metric, k, signature, cb_fpr)</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>expr_mtx <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">as.matrix</span>(adata<span class="sc">$</span>raw<span class="sc">$</span>X))</span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(expr_mtx) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(adata<span class="sc">$</span>X)</span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(expr_mtx) <span class="ot">&lt;-</span> adata<span class="sc">$</span>var_names</span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>srt <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(</span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>  expr_mtx,</span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">"RNA"</span>,</span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">project =</span> <span class="st">"romanov2020_Hypoth_dev"</span>,</span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">meta.data =</span> <span class="fu">as.data.frame</span>(adata<span class="sc">$</span>obs),</span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">row.names =</span> <span class="fu">colnames</span>(adata<span class="sc">$</span>X)</span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a>regmtx <span class="ot">&lt;-</span> <span class="fu">t</span>(adata<span class="sc">$</span>obsm<span class="sc">$</span>X_aucell)</span>
<span id="cb15-23"><a href="#cb15-23" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(regmtx) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(srt)</span>
<span id="cb15-24"><a href="#cb15-24" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(regmtx) <span class="ot">&lt;-</span> adata<span class="sc">$</span><span class="fu">obs_keys</span>() <span class="sc">%&gt;%</span> .[<span class="fu">str_detect</span>(<span class="at">string =</span> ., <span class="at">pattern =</span> <span class="st">"^Regulon"</span>)]</span>
<span id="cb15-25"><a href="#cb15-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-26"><a href="#cb15-26" aria-hidden="true" tabindex="-1"></a>regsrt <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(</span>
<span id="cb15-27"><a href="#cb15-27" aria-hidden="true" tabindex="-1"></a>  regmtx,</span>
<span id="cb15-28"><a href="#cb15-28" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">"Regulons"</span>,</span>
<span id="cb15-29"><a href="#cb15-29" aria-hidden="true" tabindex="-1"></a>  <span class="at">project =</span> <span class="st">"romanov2020_Hypoth_dev"</span>,</span>
<span id="cb15-30"><a href="#cb15-30" aria-hidden="true" tabindex="-1"></a>  <span class="at">meta.data =</span> <span class="fu">as.data.frame</span>(adata<span class="sc">$</span>obs)</span>
<span id="cb15-31"><a href="#cb15-31" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb15-32"><a href="#cb15-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-33"><a href="#cb15-33" aria-hidden="true" tabindex="-1"></a>densmap_logit <span class="ot">&lt;-</span> adata<span class="sc">$</span>obsm<span class="sc">$</span>X_umap_logit</span>
<span id="cb15-34"><a href="#cb15-34" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(densmap_logit) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"densMAP_logit_1"</span>, <span class="st">"densMAP_logit_2"</span>)</span>
<span id="cb15-35"><a href="#cb15-35" aria-hidden="true" tabindex="-1"></a>srt[[<span class="st">"densmap_logit"</span>]] <span class="ot">&lt;-</span> <span class="fu">CreateDimReducObject</span>(<span class="at">embeddings =</span> densmap_logit, <span class="at">key =</span> <span class="st">"densMAP_logit_"</span>, <span class="at">assay =</span> <span class="fu">DefaultAssay</span>(srt))</span>
<span id="cb15-36"><a href="#cb15-36" aria-hidden="true" tabindex="-1"></a><span class="co"># srt &lt;- ProjectDim(srt, reduction = "densmap_logit")</span></span>
<span id="cb15-37"><a href="#cb15-37" aria-hidden="true" tabindex="-1"></a>regsrt[[<span class="st">"densmap_logit"</span>]] <span class="ot">&lt;-</span> <span class="fu">CreateDimReducObject</span>(<span class="at">embeddings =</span> densmap_logit, <span class="at">key =</span> <span class="st">"densMAP_logit_"</span>, <span class="at">assay =</span> <span class="fu">DefaultAssay</span>(regsrt))</span>
<span id="cb15-38"><a href="#cb15-38" aria-hidden="true" tabindex="-1"></a><span class="co"># regsrt &lt;- ProjectDim(regsrt, reduction = "densmap_logit")</span></span>
<span id="cb15-39"><a href="#cb15-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-40"><a href="#cb15-40" aria-hidden="true" tabindex="-1"></a>densmap_xgboost <span class="ot">&lt;-</span> adata<span class="sc">$</span>obsm<span class="sc">$</span>X_umap</span>
<span id="cb15-41"><a href="#cb15-41" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(densmap_xgboost) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"densMAP_xgboost_1"</span>, <span class="st">"densMAP_xgboost_2"</span>)</span>
<span id="cb15-42"><a href="#cb15-42" aria-hidden="true" tabindex="-1"></a>srt[[<span class="st">"densmap_xgboost"</span>]] <span class="ot">&lt;-</span> <span class="fu">CreateDimReducObject</span>(<span class="at">embeddings =</span> densmap_xgboost, <span class="at">key =</span> <span class="st">"densMAP_xgboost_"</span>, <span class="at">assay =</span> <span class="fu">DefaultAssay</span>(srt))</span>
<span id="cb15-43"><a href="#cb15-43" aria-hidden="true" tabindex="-1"></a><span class="co"># srt &lt;- ProjectDim(srt, reduction = "densmap_xgboost")</span></span>
<span id="cb15-44"><a href="#cb15-44" aria-hidden="true" tabindex="-1"></a>regsrt[[<span class="st">"densmap_xgboost"</span>]] <span class="ot">&lt;-</span> <span class="fu">CreateDimReducObject</span>(<span class="at">embeddings =</span> densmap_xgboost, <span class="at">key =</span> <span class="st">"densMAP_xgboost_"</span>, <span class="at">assay =</span> <span class="fu">DefaultAssay</span>(regsrt))</span>
<span id="cb15-45"><a href="#cb15-45" aria-hidden="true" tabindex="-1"></a><span class="co"># regsrt &lt;- ProjectDim(regsrt, reduction = "densmap_xgboost")</span></span>
<span id="cb15-46"><a href="#cb15-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-47"><a href="#cb15-47" aria-hidden="true" tabindex="-1"></a><span class="fu">Idents</span>(srt) <span class="ot">&lt;-</span> <span class="st">"predict_xgboost_subregion"</span></span>
<span id="cb15-48"><a href="#cb15-48" aria-hidden="true" tabindex="-1"></a><span class="fu">Idents</span>(regsrt) <span class="ot">&lt;-</span> <span class="st">"predict_xgboost_subregion"</span></span>
<span id="cb15-49"><a href="#cb15-49" aria-hidden="true" tabindex="-1"></a>srt <span class="ot">&lt;-</span> <span class="fu">Store_Palette_Seurat</span>(<span class="at">seurat_object =</span> srt, <span class="at">palette =</span> <span class="fu">rev</span>(<span class="fu">brewer.pal</span>(<span class="at">n =</span> <span class="dv">11</span>, <span class="at">name =</span> <span class="st">"Spectral"</span>)), <span class="at">palette_name =</span> <span class="st">"expr_Colour_Pal"</span>)</span>
<span id="cb15-50"><a href="#cb15-50" aria-hidden="true" tabindex="-1"></a>regsrt <span class="ot">&lt;-</span> <span class="fu">Store_Palette_Seurat</span>(<span class="at">seurat_object =</span> regsrt, <span class="at">palette =</span> <span class="fu">rev</span>(<span class="fu">brewer.pal</span>(<span class="at">n =</span> <span class="dv">11</span>, <span class="at">name =</span> <span class="st">"PiYG"</span>)), <span class="at">palette_name =</span> <span class="st">"expr_Colour_Pal"</span>)</span>
<span id="cb15-51"><a href="#cb15-51" aria-hidden="true" tabindex="-1"></a><span class="fu">SaveH5Seurat</span>(srt, <span class="at">filename =</span> srt_path, <span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-52"><a href="#cb15-52" aria-hidden="true" tabindex="-1"></a><span class="fu">SaveH5Seurat</span>(regsrt, <span class="at">filename =</span> regsrt_path, <span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-layout-align="center">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>srt <span class="ot">&lt;-</span> <span class="fu">LoadH5Seurat</span>(<span class="at">file =</span> srt_path)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>regsrt <span class="ot">&lt;-</span> <span class="fu">LoadH5Seurat</span>(<span class="at">file =</span> regsrt_path)</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="fu">Idents</span>(srt) <span class="ot">&lt;-</span> <span class="st">"predict_xgboost_subregion"</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="fu">Idents</span>(regsrt) <span class="ot">&lt;-</span> <span class="st">"predict_xgboost_subregion"</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(srt)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>An object of class Seurat 
21989 features across 3042 samples within 1 assay 
Active assay: RNA (21989 features, 0 variable features)
 2 dimensional reductions calculated: densmap_logit, densmap_xgboost</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(regsrt)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>An object of class Seurat 
52 features across 3042 samples within 1 assay 
Active assay: Regulons (52 features, 0 variable features)
 2 dimensional reductions calculated: densmap_logit, densmap_xgboost</code></pre>
</div>
</div>
<div class="cell" data-layout-align="center">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>srt <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(srt)</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>srt <span class="ot">&lt;-</span> <span class="fu">FindVariableFeatures</span>(srt, <span class="at">selection.method =</span> <span class="st">"vst"</span>, <span class="at">nfeatures =</span> <span class="dv">3000</span>)</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>all.genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(srt)</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>srt <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(srt, <span class="at">features =</span> all.genes)</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>all_markers_genes <span class="ot">&lt;-</span> <span class="fu">FindAllMarkers</span>(<span class="at">object =</span> srt, <span class="at">verbose =</span> F, <span class="at">test.use =</span> <span class="st">"MAST"</span>, <span class="at">only.pos =</span> T, <span class="at">min.pct =</span> <span class="fl">0.05</span>, <span class="at">logfc.threshold =</span> <span class="fl">0.15</span>) <span class="sc">%&gt;%</span></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Add_Pct_Diff</span>()</span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(all_markers_genes, <span class="fu">here</span>(data_dir, <span class="fu">sprintf</span>(<span class="st">"resolved_subregions_by_microclusters/%s-all-marker-genes.csv"</span>, bioproject)))</span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>all_markers_genes <span class="ot">&lt;-</span> all_markers_genes <span class="sc">%&gt;%</span></span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(pct_diff <span class="sc">&gt;</span> <span class="fl">0.1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span></span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(avg_log2FC)) <span class="sc">%&gt;%</span></span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(cluster)</span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a>all_markers_genes2 <span class="ot">&lt;-</span> all_markers_genes <span class="sc">%&gt;%</span></span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(gene <span class="sc">%in%</span> <span class="fu">c</span>(npr, nmr, transcription_factors)) <span class="sc">%&gt;%</span></span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span></span>
<span id="cb20-18"><a href="#cb20-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(avg_log2FC)) <span class="sc">%&gt;%</span></span>
<span id="cb20-19"><a href="#cb20-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(cluster)</span>
<span id="cb20-20"><a href="#cb20-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-21"><a href="#cb20-21" aria-hidden="true" tabindex="-1"></a>top_5 <span class="ot">&lt;-</span> <span class="fu">Extract_Top_Markers</span>(<span class="at">marker_dataframe =</span> all_markers_genes, <span class="at">num_genes =</span> <span class="dv">5</span>, <span class="at">group_by =</span> <span class="st">"cluster"</span>, <span class="at">gene_column =</span> <span class="st">"gene"</span>, <span class="at">rank_by =</span> <span class="st">"avg_log2FC"</span>, <span class="at">make_unique =</span> <span class="cn">TRUE</span>, <span class="at">named_vector =</span> <span class="cn">FALSE</span>)</span>
<span id="cb20-22"><a href="#cb20-22" aria-hidden="true" tabindex="-1"></a>top_30 <span class="ot">&lt;-</span> <span class="fu">Extract_Top_Markers</span>(<span class="at">marker_dataframe =</span> all_markers_genes, <span class="at">num_genes =</span> <span class="dv">30</span>, <span class="at">group_by =</span> <span class="st">"cluster"</span>, <span class="at">gene_column =</span> <span class="st">"gene"</span>, <span class="at">rank_by =</span> <span class="st">"avg_log2FC"</span>)</span>
<span id="cb20-23"><a href="#cb20-23" aria-hidden="true" tabindex="-1"></a>top_5s <span class="ot">&lt;-</span> <span class="fu">Extract_Top_Markers</span>(<span class="at">marker_dataframe =</span> all_markers_genes2, <span class="at">num_genes =</span> <span class="dv">5</span>, <span class="at">group_by =</span> <span class="st">"cluster"</span>, <span class="at">gene_column =</span> <span class="st">"gene"</span>, <span class="at">rank_by =</span> <span class="st">"avg_log2FC"</span>, <span class="at">make_unique =</span> <span class="cn">TRUE</span>, <span class="at">named_vector =</span> <span class="cn">FALSE</span>)</span>
<span id="cb20-24"><a href="#cb20-24" aria-hidden="true" tabindex="-1"></a>top_30s <span class="ot">&lt;-</span> <span class="fu">Extract_Top_Markers</span>(<span class="at">marker_dataframe =</span> all_markers_genes2, <span class="at">num_genes =</span> <span class="dv">30</span>, <span class="at">group_by =</span> <span class="st">"cluster"</span>, <span class="at">gene_column =</span> <span class="st">"gene"</span>, <span class="at">rank_by =</span> <span class="st">"avg_log2FC"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-layout-align="center">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">Iterate_FeaturePlot_scCustom</span>(<span class="at">reduction =</span> <span class="st">"densmap_xgboost"</span>,</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">na_cutoff =</span> <span class="cn">NA</span>,</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">pt.size =</span> <span class="dv">4</span>,</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">order =</span> <span class="cn">TRUE</span>,</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">alpha_na_exp =</span> <span class="fl">0.1</span>,</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">alpha_exp =</span> <span class="fl">0.75</span>, <span class="at">seurat_object =</span> srt, <span class="at">gene_list =</span> top_30, <span class="at">single_pdf =</span> <span class="cn">FALSE</span>, <span class="at">file_path =</span> plots_dir, <span class="at">file_type=</span><span class="st">".pdf"</span>, <span class="at">file_name =</span> <span class="fu">sprintf</span>(<span class="st">"%s.pdf"</span>, bioproject), <span class="at">colors_use =</span> srt<span class="sc">@</span>misc<span class="sc">$</span>expr_Colour_Pal</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-layout-align="center">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">Clustered_DotPlot</span>(<span class="at">seurat_object =</span> srt, <span class="at">colors_use_exp =</span> <span class="fu">viridis</span>(<span class="at">n =</span> <span class="dv">30</span>, <span class="at">alpha =</span> .<span class="dv">75</span>, <span class="at">direction =</span> <span class="sc">-</span><span class="dv">1</span>, <span class="at">option =</span> <span class="st">"E"</span>), <span class="at">features =</span> top_5, <span class="at">k =</span> <span class="dv">9</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="04-de_test-dendrogram_files/figure-html/plt-dotplot-dendrogram-genes-1.png" class="img-fluid figure-img" width="3600"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[[1]]</code></pre>
</div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="04-de_test-dendrogram_files/figure-html/plt-dotplot-dendrogram-genes-2.png" class="img-fluid figure-img" width="3600"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
[[2]]</code></pre>
</div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="04-de_test-dendrogram_files/figure-html/plt-dotplot-dendrogram-genes-3.png" class="img-fluid figure-img" width="3600"></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-layout-align="center">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">Clustered_DotPlot</span>(<span class="at">seurat_object =</span> srt, <span class="at">colors_use_exp =</span> <span class="fu">viridis</span>(<span class="at">n =</span> <span class="dv">30</span>, <span class="at">alpha =</span> .<span class="dv">75</span>, <span class="at">direction =</span> <span class="sc">-</span><span class="dv">1</span>, <span class="at">option =</span> <span class="st">"E"</span>), <span class="at">features =</span> top_5s, <span class="at">k =</span> <span class="dv">9</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="04-de_test-dendrogram_files/figure-html/plt-dotplot-dendrogram-sgenes-1.png" class="img-fluid figure-img" width="3600"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[[1]]</code></pre>
</div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="04-de_test-dendrogram_files/figure-html/plt-dotplot-dendrogram-sgenes-2.png" class="img-fluid figure-img" width="3600"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
[[2]]</code></pre>
</div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="04-de_test-dendrogram_files/figure-html/plt-dotplot-dendrogram-sgenes-3.png" class="img-fluid figure-img" width="3600"></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-layout-align="center">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DotPlot_scCustom</span>(<span class="at">seurat_object =</span> srt, <span class="at">colors_use =</span> <span class="fu">viridis</span>(<span class="at">n =</span> <span class="dv">30</span>, <span class="at">alpha =</span> .<span class="dv">75</span>, <span class="at">direction =</span> <span class="sc">-</span><span class="dv">1</span>, <span class="at">option =</span> <span class="st">"E"</span>), <span class="at">features =</span> npr[npr <span class="sc">%in%</span> <span class="fu">rownames</span>(srt)], <span class="at">flip_axes =</span> T, <span class="at">x_lab_rotate =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="04-de_test-dendrogram_files/figure-html/plt-dotplot-dendrogram-genes-npr-1.png" class="img-fluid figure-img" width="3600"></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-layout-align="center">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DotPlot_scCustom</span>(<span class="at">seurat_object =</span> srt, <span class="at">colors_use =</span> <span class="fu">viridis</span>(<span class="at">n =</span> <span class="dv">30</span>, <span class="at">alpha =</span> .<span class="dv">75</span>, <span class="at">direction =</span> <span class="sc">-</span><span class="dv">1</span>, <span class="at">option =</span> <span class="st">"E"</span>), <span class="at">features =</span> nmr[nmr <span class="sc">%in%</span> <span class="fu">rownames</span>(srt)], <span class="at">flip_axes =</span> T, <span class="at">x_lab_rotate =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="04-de_test-dendrogram_files/figure-html/plt-dotplot-dendrogram-genes-nmr-1.png" class="img-fluid figure-img" width="3600"></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-layout-align="center">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DotPlot_scCustom</span>(<span class="at">seurat_object =</span> srt, <span class="at">colors_use =</span> <span class="fu">viridis</span>(<span class="at">n =</span> <span class="dv">30</span>, <span class="at">alpha =</span> .<span class="dv">75</span>, <span class="at">direction =</span> <span class="sc">-</span><span class="dv">1</span>, <span class="at">option =</span> <span class="st">"E"</span>), <span class="at">features =</span> genes.embed[genes.embed <span class="sc">%in%</span> <span class="fu">rownames</span>(srt)], <span class="at">flip_axes =</span> T, <span class="at">x_lab_rotate =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="04-de_test-dendrogram_files/figure-html/plt-dotplot-dendrogram-genes-adgen-1.png" class="img-fluid figure-img" width="3600"></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-layout-align="center">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DotPlot_scCustom</span>(<span class="at">seurat_object =</span> srt, <span class="at">colors_use =</span> <span class="fu">viridis</span>(<span class="at">n =</span> <span class="dv">30</span>, <span class="at">alpha =</span> .<span class="dv">75</span>, <span class="at">direction =</span> <span class="sc">-</span><span class="dv">1</span>, <span class="at">option =</span> <span class="st">"E"</span>), <span class="at">features =</span> mitochondrial[mitochondrial <span class="sc">%in%</span> <span class="fu">rownames</span>(srt)], <span class="at">flip_axes =</span> T, <span class="at">x_lab_rotate =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="04-de_test-dendrogram_files/figure-html/plt-dotplot-dendrogram-genes-mitochondrial-1.png" class="img-fluid figure-img" width="3600"></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-layout-align="center">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>de_01 <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(srt,</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">"RNA"</span>, <span class="at">ident.1 =</span> <span class="st">"0"</span>, <span class="at">ident.2 =</span> <span class="st">"1"</span>,</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">test.use =</span> <span class="st">"LR"</span>,</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">only.pos =</span> F,</span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">min.pct =</span> <span class="fl">0.01</span></span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>)</span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a>de_02 <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(srt,</span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">"RNA"</span>, <span class="at">ident.1 =</span> <span class="st">"0"</span>, <span class="at">ident.2 =</span> <span class="st">"2"</span>,</span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">test.use =</span> <span class="st">"LR"</span>,</span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">only.pos =</span> F,</span>
<span id="cb32-12"><a href="#cb32-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">min.pct =</span> <span class="fl">0.01</span></span>
<span id="cb32-13"><a href="#cb32-13" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>)</span>
<span id="cb32-14"><a href="#cb32-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-15"><a href="#cb32-15" aria-hidden="true" tabindex="-1"></a>de_03 <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(srt,</span>
<span id="cb32-16"><a href="#cb32-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">"RNA"</span>, <span class="at">ident.1 =</span> <span class="st">"0"</span>, <span class="at">ident.2 =</span> <span class="st">"3"</span>,</span>
<span id="cb32-17"><a href="#cb32-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">test.use =</span> <span class="st">"LR"</span>,</span>
<span id="cb32-18"><a href="#cb32-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">only.pos =</span> F,</span>
<span id="cb32-19"><a href="#cb32-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">min.pct =</span> <span class="fl">0.01</span></span>
<span id="cb32-20"><a href="#cb32-20" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>)</span>
<span id="cb32-21"><a href="#cb32-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-22"><a href="#cb32-22" aria-hidden="true" tabindex="-1"></a>de_04 <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(srt,</span>
<span id="cb32-23"><a href="#cb32-23" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">"RNA"</span>, <span class="at">ident.1 =</span> <span class="st">"0"</span>, <span class="at">ident.2 =</span> <span class="st">"4"</span>,</span>
<span id="cb32-24"><a href="#cb32-24" aria-hidden="true" tabindex="-1"></a>  <span class="at">test.use =</span> <span class="st">"LR"</span>,</span>
<span id="cb32-25"><a href="#cb32-25" aria-hidden="true" tabindex="-1"></a>  <span class="at">only.pos =</span> F,</span>
<span id="cb32-26"><a href="#cb32-26" aria-hidden="true" tabindex="-1"></a>  <span class="at">min.pct =</span> <span class="fl">0.01</span></span>
<span id="cb32-27"><a href="#cb32-27" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>)</span>
<span id="cb32-28"><a href="#cb32-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-29"><a href="#cb32-29" aria-hidden="true" tabindex="-1"></a>de_05 <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(srt,</span>
<span id="cb32-30"><a href="#cb32-30" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">"RNA"</span>, <span class="at">ident.1 =</span> <span class="st">"0"</span>, <span class="at">ident.2 =</span> <span class="st">"5"</span>,</span>
<span id="cb32-31"><a href="#cb32-31" aria-hidden="true" tabindex="-1"></a>  <span class="at">test.use =</span> <span class="st">"LR"</span>,</span>
<span id="cb32-32"><a href="#cb32-32" aria-hidden="true" tabindex="-1"></a>  <span class="at">only.pos =</span> F,</span>
<span id="cb32-33"><a href="#cb32-33" aria-hidden="true" tabindex="-1"></a>  <span class="at">min.pct =</span> <span class="fl">0.01</span></span>
<span id="cb32-34"><a href="#cb32-34" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>)</span>
<span id="cb32-35"><a href="#cb32-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-36"><a href="#cb32-36" aria-hidden="true" tabindex="-1"></a>de_06 <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(srt,</span>
<span id="cb32-37"><a href="#cb32-37" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">"RNA"</span>, <span class="at">ident.1 =</span> <span class="st">"0"</span>, <span class="at">ident.2 =</span> <span class="st">"6"</span>,</span>
<span id="cb32-38"><a href="#cb32-38" aria-hidden="true" tabindex="-1"></a>  <span class="at">test.use =</span> <span class="st">"LR"</span>,</span>
<span id="cb32-39"><a href="#cb32-39" aria-hidden="true" tabindex="-1"></a>  <span class="at">only.pos =</span> F,</span>
<span id="cb32-40"><a href="#cb32-40" aria-hidden="true" tabindex="-1"></a>  <span class="at">min.pct =</span> <span class="fl">0.01</span></span>
<span id="cb32-41"><a href="#cb32-41" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>)</span>
<span id="cb32-42"><a href="#cb32-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-43"><a href="#cb32-43" aria-hidden="true" tabindex="-1"></a>de_07 <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(srt,</span>
<span id="cb32-44"><a href="#cb32-44" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">"RNA"</span>, <span class="at">ident.1 =</span> <span class="st">"0"</span>, <span class="at">ident.2 =</span> <span class="st">"7"</span>,</span>
<span id="cb32-45"><a href="#cb32-45" aria-hidden="true" tabindex="-1"></a>  <span class="at">test.use =</span> <span class="st">"LR"</span>,</span>
<span id="cb32-46"><a href="#cb32-46" aria-hidden="true" tabindex="-1"></a>  <span class="at">only.pos =</span> F,</span>
<span id="cb32-47"><a href="#cb32-47" aria-hidden="true" tabindex="-1"></a>  <span class="at">min.pct =</span> <span class="fl">0.01</span></span>
<span id="cb32-48"><a href="#cb32-48" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>)</span>
<span id="cb32-49"><a href="#cb32-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-50"><a href="#cb32-50" aria-hidden="true" tabindex="-1"></a>de_12 <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(srt,</span>
<span id="cb32-51"><a href="#cb32-51" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">"RNA"</span>, <span class="at">ident.1 =</span> <span class="st">"1"</span>, <span class="at">ident.2 =</span> <span class="st">"2"</span>,</span>
<span id="cb32-52"><a href="#cb32-52" aria-hidden="true" tabindex="-1"></a>  <span class="at">test.use =</span> <span class="st">"LR"</span>,</span>
<span id="cb32-53"><a href="#cb32-53" aria-hidden="true" tabindex="-1"></a>  <span class="at">only.pos =</span> F,</span>
<span id="cb32-54"><a href="#cb32-54" aria-hidden="true" tabindex="-1"></a>  <span class="at">min.pct =</span> <span class="fl">0.01</span></span>
<span id="cb32-55"><a href="#cb32-55" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>)</span>
<span id="cb32-56"><a href="#cb32-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-57"><a href="#cb32-57" aria-hidden="true" tabindex="-1"></a>de_13 <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(srt,</span>
<span id="cb32-58"><a href="#cb32-58" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">"RNA"</span>, <span class="at">ident.1 =</span> <span class="st">"1"</span>, <span class="at">ident.2 =</span> <span class="st">"3"</span>,</span>
<span id="cb32-59"><a href="#cb32-59" aria-hidden="true" tabindex="-1"></a>  <span class="at">test.use =</span> <span class="st">"LR"</span>,</span>
<span id="cb32-60"><a href="#cb32-60" aria-hidden="true" tabindex="-1"></a>  <span class="at">only.pos =</span> F,</span>
<span id="cb32-61"><a href="#cb32-61" aria-hidden="true" tabindex="-1"></a>  <span class="at">min.pct =</span> <span class="fl">0.01</span></span>
<span id="cb32-62"><a href="#cb32-62" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>)</span>
<span id="cb32-63"><a href="#cb32-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-64"><a href="#cb32-64" aria-hidden="true" tabindex="-1"></a>de_14 <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(srt,</span>
<span id="cb32-65"><a href="#cb32-65" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">"RNA"</span>, <span class="at">ident.1 =</span> <span class="st">"1"</span>, <span class="at">ident.2 =</span> <span class="st">"4"</span>,</span>
<span id="cb32-66"><a href="#cb32-66" aria-hidden="true" tabindex="-1"></a>  <span class="at">test.use =</span> <span class="st">"LR"</span>,</span>
<span id="cb32-67"><a href="#cb32-67" aria-hidden="true" tabindex="-1"></a>  <span class="at">only.pos =</span> F,</span>
<span id="cb32-68"><a href="#cb32-68" aria-hidden="true" tabindex="-1"></a>  <span class="at">min.pct =</span> <span class="fl">0.01</span></span>
<span id="cb32-69"><a href="#cb32-69" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>)</span>
<span id="cb32-70"><a href="#cb32-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-71"><a href="#cb32-71" aria-hidden="true" tabindex="-1"></a>de_15 <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(srt,</span>
<span id="cb32-72"><a href="#cb32-72" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">"RNA"</span>, <span class="at">ident.1 =</span> <span class="st">"1"</span>, <span class="at">ident.2 =</span> <span class="st">"5"</span>,</span>
<span id="cb32-73"><a href="#cb32-73" aria-hidden="true" tabindex="-1"></a>  <span class="at">test.use =</span> <span class="st">"LR"</span>,</span>
<span id="cb32-74"><a href="#cb32-74" aria-hidden="true" tabindex="-1"></a>  <span class="at">only.pos =</span> F,</span>
<span id="cb32-75"><a href="#cb32-75" aria-hidden="true" tabindex="-1"></a>  <span class="at">min.pct =</span> <span class="fl">0.01</span></span>
<span id="cb32-76"><a href="#cb32-76" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>)</span>
<span id="cb32-77"><a href="#cb32-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-78"><a href="#cb32-78" aria-hidden="true" tabindex="-1"></a>de_16 <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(srt,</span>
<span id="cb32-79"><a href="#cb32-79" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">"RNA"</span>, <span class="at">ident.1 =</span> <span class="st">"1"</span>, <span class="at">ident.2 =</span> <span class="st">"6"</span>,</span>
<span id="cb32-80"><a href="#cb32-80" aria-hidden="true" tabindex="-1"></a>  <span class="at">test.use =</span> <span class="st">"LR"</span>,</span>
<span id="cb32-81"><a href="#cb32-81" aria-hidden="true" tabindex="-1"></a>  <span class="at">only.pos =</span> F,</span>
<span id="cb32-82"><a href="#cb32-82" aria-hidden="true" tabindex="-1"></a>  <span class="at">min.pct =</span> <span class="fl">0.01</span></span>
<span id="cb32-83"><a href="#cb32-83" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>)</span>
<span id="cb32-84"><a href="#cb32-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-85"><a href="#cb32-85" aria-hidden="true" tabindex="-1"></a>de_17 <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(srt,</span>
<span id="cb32-86"><a href="#cb32-86" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">"RNA"</span>, <span class="at">ident.1 =</span> <span class="st">"1"</span>, <span class="at">ident.2 =</span> <span class="st">"7"</span>,</span>
<span id="cb32-87"><a href="#cb32-87" aria-hidden="true" tabindex="-1"></a>  <span class="at">test.use =</span> <span class="st">"LR"</span>,</span>
<span id="cb32-88"><a href="#cb32-88" aria-hidden="true" tabindex="-1"></a>  <span class="at">only.pos =</span> F,</span>
<span id="cb32-89"><a href="#cb32-89" aria-hidden="true" tabindex="-1"></a>  <span class="at">min.pct =</span> <span class="fl">0.01</span></span>
<span id="cb32-90"><a href="#cb32-90" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>)</span>
<span id="cb32-91"><a href="#cb32-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-92"><a href="#cb32-92" aria-hidden="true" tabindex="-1"></a>de_23 <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(srt,</span>
<span id="cb32-93"><a href="#cb32-93" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">"RNA"</span>, <span class="at">ident.1 =</span> <span class="st">"2"</span>, <span class="at">ident.2 =</span> <span class="st">"3"</span>,</span>
<span id="cb32-94"><a href="#cb32-94" aria-hidden="true" tabindex="-1"></a>  <span class="at">test.use =</span> <span class="st">"LR"</span>,</span>
<span id="cb32-95"><a href="#cb32-95" aria-hidden="true" tabindex="-1"></a>  <span class="at">only.pos =</span> F,</span>
<span id="cb32-96"><a href="#cb32-96" aria-hidden="true" tabindex="-1"></a>  <span class="at">min.pct =</span> <span class="fl">0.01</span></span>
<span id="cb32-97"><a href="#cb32-97" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>)</span>
<span id="cb32-98"><a href="#cb32-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-99"><a href="#cb32-99" aria-hidden="true" tabindex="-1"></a>de_24 <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(srt,</span>
<span id="cb32-100"><a href="#cb32-100" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">"RNA"</span>, <span class="at">ident.1 =</span> <span class="st">"2"</span>, <span class="at">ident.2 =</span> <span class="st">"4"</span>,</span>
<span id="cb32-101"><a href="#cb32-101" aria-hidden="true" tabindex="-1"></a>  <span class="at">test.use =</span> <span class="st">"LR"</span>,</span>
<span id="cb32-102"><a href="#cb32-102" aria-hidden="true" tabindex="-1"></a>  <span class="at">only.pos =</span> F,</span>
<span id="cb32-103"><a href="#cb32-103" aria-hidden="true" tabindex="-1"></a>  <span class="at">min.pct =</span> <span class="fl">0.01</span></span>
<span id="cb32-104"><a href="#cb32-104" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>)</span>
<span id="cb32-105"><a href="#cb32-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-106"><a href="#cb32-106" aria-hidden="true" tabindex="-1"></a>de_25 <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(srt,</span>
<span id="cb32-107"><a href="#cb32-107" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">"RNA"</span>, <span class="at">ident.1 =</span> <span class="st">"2"</span>, <span class="at">ident.2 =</span> <span class="st">"5"</span>,</span>
<span id="cb32-108"><a href="#cb32-108" aria-hidden="true" tabindex="-1"></a>  <span class="at">test.use =</span> <span class="st">"LR"</span>,</span>
<span id="cb32-109"><a href="#cb32-109" aria-hidden="true" tabindex="-1"></a>  <span class="at">only.pos =</span> F,</span>
<span id="cb32-110"><a href="#cb32-110" aria-hidden="true" tabindex="-1"></a>  <span class="at">min.pct =</span> <span class="fl">0.01</span></span>
<span id="cb32-111"><a href="#cb32-111" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>)</span>
<span id="cb32-112"><a href="#cb32-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-113"><a href="#cb32-113" aria-hidden="true" tabindex="-1"></a>de_26 <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(srt,</span>
<span id="cb32-114"><a href="#cb32-114" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">"RNA"</span>, <span class="at">ident.1 =</span> <span class="st">"2"</span>, <span class="at">ident.2 =</span> <span class="st">"6"</span>,</span>
<span id="cb32-115"><a href="#cb32-115" aria-hidden="true" tabindex="-1"></a>  <span class="at">test.use =</span> <span class="st">"LR"</span>,</span>
<span id="cb32-116"><a href="#cb32-116" aria-hidden="true" tabindex="-1"></a>  <span class="at">only.pos =</span> F,</span>
<span id="cb32-117"><a href="#cb32-117" aria-hidden="true" tabindex="-1"></a>  <span class="at">min.pct =</span> <span class="fl">0.01</span></span>
<span id="cb32-118"><a href="#cb32-118" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>)</span>
<span id="cb32-119"><a href="#cb32-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-120"><a href="#cb32-120" aria-hidden="true" tabindex="-1"></a>de_27 <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(srt,</span>
<span id="cb32-121"><a href="#cb32-121" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">"RNA"</span>, <span class="at">ident.1 =</span> <span class="st">"2"</span>, <span class="at">ident.2 =</span> <span class="st">"7"</span>,</span>
<span id="cb32-122"><a href="#cb32-122" aria-hidden="true" tabindex="-1"></a>  <span class="at">test.use =</span> <span class="st">"LR"</span>,</span>
<span id="cb32-123"><a href="#cb32-123" aria-hidden="true" tabindex="-1"></a>  <span class="at">only.pos =</span> F,</span>
<span id="cb32-124"><a href="#cb32-124" aria-hidden="true" tabindex="-1"></a>  <span class="at">min.pct =</span> <span class="fl">0.01</span></span>
<span id="cb32-125"><a href="#cb32-125" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>)</span>
<span id="cb32-126"><a href="#cb32-126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-127"><a href="#cb32-127" aria-hidden="true" tabindex="-1"></a>de_34 <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(srt,</span>
<span id="cb32-128"><a href="#cb32-128" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">"RNA"</span>, <span class="at">ident.1 =</span> <span class="st">"3"</span>, <span class="at">ident.2 =</span> <span class="st">"4"</span>,</span>
<span id="cb32-129"><a href="#cb32-129" aria-hidden="true" tabindex="-1"></a>  <span class="at">test.use =</span> <span class="st">"LR"</span>,</span>
<span id="cb32-130"><a href="#cb32-130" aria-hidden="true" tabindex="-1"></a>  <span class="at">only.pos =</span> F,</span>
<span id="cb32-131"><a href="#cb32-131" aria-hidden="true" tabindex="-1"></a>  <span class="at">min.pct =</span> <span class="fl">0.01</span></span>
<span id="cb32-132"><a href="#cb32-132" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>)</span>
<span id="cb32-133"><a href="#cb32-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-134"><a href="#cb32-134" aria-hidden="true" tabindex="-1"></a>de_35 <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(srt,</span>
<span id="cb32-135"><a href="#cb32-135" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">"RNA"</span>, <span class="at">ident.1 =</span> <span class="st">"3"</span>, <span class="at">ident.2 =</span> <span class="st">"5"</span>,</span>
<span id="cb32-136"><a href="#cb32-136" aria-hidden="true" tabindex="-1"></a>  <span class="at">test.use =</span> <span class="st">"LR"</span>,</span>
<span id="cb32-137"><a href="#cb32-137" aria-hidden="true" tabindex="-1"></a>  <span class="at">only.pos =</span> F,</span>
<span id="cb32-138"><a href="#cb32-138" aria-hidden="true" tabindex="-1"></a>  <span class="at">min.pct =</span> <span class="fl">0.01</span></span>
<span id="cb32-139"><a href="#cb32-139" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>)</span>
<span id="cb32-140"><a href="#cb32-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-141"><a href="#cb32-141" aria-hidden="true" tabindex="-1"></a>de_36 <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(srt,</span>
<span id="cb32-142"><a href="#cb32-142" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">"RNA"</span>, <span class="at">ident.1 =</span> <span class="st">"3"</span>, <span class="at">ident.2 =</span> <span class="st">"6"</span>,</span>
<span id="cb32-143"><a href="#cb32-143" aria-hidden="true" tabindex="-1"></a>  <span class="at">test.use =</span> <span class="st">"LR"</span>,</span>
<span id="cb32-144"><a href="#cb32-144" aria-hidden="true" tabindex="-1"></a>  <span class="at">only.pos =</span> F,</span>
<span id="cb32-145"><a href="#cb32-145" aria-hidden="true" tabindex="-1"></a>  <span class="at">min.pct =</span> <span class="fl">0.01</span></span>
<span id="cb32-146"><a href="#cb32-146" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>)</span>
<span id="cb32-147"><a href="#cb32-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-148"><a href="#cb32-148" aria-hidden="true" tabindex="-1"></a>de_37 <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(srt,</span>
<span id="cb32-149"><a href="#cb32-149" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">"RNA"</span>, <span class="at">ident.1 =</span> <span class="st">"3"</span>, <span class="at">ident.2 =</span> <span class="st">"7"</span>,</span>
<span id="cb32-150"><a href="#cb32-150" aria-hidden="true" tabindex="-1"></a>  <span class="at">test.use =</span> <span class="st">"LR"</span>,</span>
<span id="cb32-151"><a href="#cb32-151" aria-hidden="true" tabindex="-1"></a>  <span class="at">only.pos =</span> F,</span>
<span id="cb32-152"><a href="#cb32-152" aria-hidden="true" tabindex="-1"></a>  <span class="at">min.pct =</span> <span class="fl">0.01</span></span>
<span id="cb32-153"><a href="#cb32-153" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>)</span>
<span id="cb32-154"><a href="#cb32-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-155"><a href="#cb32-155" aria-hidden="true" tabindex="-1"></a>de_45 <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(srt,</span>
<span id="cb32-156"><a href="#cb32-156" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">"RNA"</span>, <span class="at">ident.1 =</span> <span class="st">"4"</span>, <span class="at">ident.2 =</span> <span class="st">"5"</span>,</span>
<span id="cb32-157"><a href="#cb32-157" aria-hidden="true" tabindex="-1"></a>  <span class="at">test.use =</span> <span class="st">"LR"</span>,</span>
<span id="cb32-158"><a href="#cb32-158" aria-hidden="true" tabindex="-1"></a>  <span class="at">only.pos =</span> F,</span>
<span id="cb32-159"><a href="#cb32-159" aria-hidden="true" tabindex="-1"></a>  <span class="at">min.pct =</span> <span class="fl">0.01</span></span>
<span id="cb32-160"><a href="#cb32-160" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>)</span>
<span id="cb32-161"><a href="#cb32-161" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-162"><a href="#cb32-162" aria-hidden="true" tabindex="-1"></a>de_46 <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(srt,</span>
<span id="cb32-163"><a href="#cb32-163" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">"RNA"</span>, <span class="at">ident.1 =</span> <span class="st">"4"</span>, <span class="at">ident.2 =</span> <span class="st">"6"</span>,</span>
<span id="cb32-164"><a href="#cb32-164" aria-hidden="true" tabindex="-1"></a>  <span class="at">test.use =</span> <span class="st">"LR"</span>,</span>
<span id="cb32-165"><a href="#cb32-165" aria-hidden="true" tabindex="-1"></a>  <span class="at">only.pos =</span> F,</span>
<span id="cb32-166"><a href="#cb32-166" aria-hidden="true" tabindex="-1"></a>  <span class="at">min.pct =</span> <span class="fl">0.01</span></span>
<span id="cb32-167"><a href="#cb32-167" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>)</span>
<span id="cb32-168"><a href="#cb32-168" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-169"><a href="#cb32-169" aria-hidden="true" tabindex="-1"></a>de_47 <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(srt,</span>
<span id="cb32-170"><a href="#cb32-170" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">"RNA"</span>, <span class="at">ident.1 =</span> <span class="st">"4"</span>, <span class="at">ident.2 =</span> <span class="st">"7"</span>,</span>
<span id="cb32-171"><a href="#cb32-171" aria-hidden="true" tabindex="-1"></a>  <span class="at">test.use =</span> <span class="st">"LR"</span>,</span>
<span id="cb32-172"><a href="#cb32-172" aria-hidden="true" tabindex="-1"></a>  <span class="at">only.pos =</span> F,</span>
<span id="cb32-173"><a href="#cb32-173" aria-hidden="true" tabindex="-1"></a>  <span class="at">min.pct =</span> <span class="fl">0.01</span></span>
<span id="cb32-174"><a href="#cb32-174" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>)</span>
<span id="cb32-175"><a href="#cb32-175" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-176"><a href="#cb32-176" aria-hidden="true" tabindex="-1"></a>de_56 <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(srt,</span>
<span id="cb32-177"><a href="#cb32-177" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">"RNA"</span>, <span class="at">ident.1 =</span> <span class="st">"5"</span>, <span class="at">ident.2 =</span> <span class="st">"6"</span>,</span>
<span id="cb32-178"><a href="#cb32-178" aria-hidden="true" tabindex="-1"></a>  <span class="at">test.use =</span> <span class="st">"LR"</span>,</span>
<span id="cb32-179"><a href="#cb32-179" aria-hidden="true" tabindex="-1"></a>  <span class="at">only.pos =</span> F,</span>
<span id="cb32-180"><a href="#cb32-180" aria-hidden="true" tabindex="-1"></a>  <span class="at">min.pct =</span> <span class="fl">0.01</span></span>
<span id="cb32-181"><a href="#cb32-181" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>)</span>
<span id="cb32-182"><a href="#cb32-182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-183"><a href="#cb32-183" aria-hidden="true" tabindex="-1"></a>de_57 <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(srt,</span>
<span id="cb32-184"><a href="#cb32-184" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">"RNA"</span>, <span class="at">ident.1 =</span> <span class="st">"5"</span>, <span class="at">ident.2 =</span> <span class="st">"7"</span>,</span>
<span id="cb32-185"><a href="#cb32-185" aria-hidden="true" tabindex="-1"></a>  <span class="at">test.use =</span> <span class="st">"LR"</span>,</span>
<span id="cb32-186"><a href="#cb32-186" aria-hidden="true" tabindex="-1"></a>  <span class="at">only.pos =</span> F,</span>
<span id="cb32-187"><a href="#cb32-187" aria-hidden="true" tabindex="-1"></a>  <span class="at">min.pct =</span> <span class="fl">0.01</span></span>
<span id="cb32-188"><a href="#cb32-188" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>)</span>
<span id="cb32-189"><a href="#cb32-189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-190"><a href="#cb32-190" aria-hidden="true" tabindex="-1"></a>de_67 <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(srt,</span>
<span id="cb32-191"><a href="#cb32-191" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">"RNA"</span>, <span class="at">ident.1 =</span> <span class="st">"6"</span>, <span class="at">ident.2 =</span> <span class="st">"7"</span>,</span>
<span id="cb32-192"><a href="#cb32-192" aria-hidden="true" tabindex="-1"></a>  <span class="at">test.use =</span> <span class="st">"LR"</span>,</span>
<span id="cb32-193"><a href="#cb32-193" aria-hidden="true" tabindex="-1"></a>  <span class="at">only.pos =</span> F,</span>
<span id="cb32-194"><a href="#cb32-194" aria-hidden="true" tabindex="-1"></a>  <span class="at">min.pct =</span> <span class="fl">0.01</span></span>
<span id="cb32-195"><a href="#cb32-195" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>)</span>
<span id="cb32-196"><a href="#cb32-196" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-197"><a href="#cb32-197" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-198"><a href="#cb32-198" aria-hidden="true" tabindex="-1"></a>top_genes <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb32-199"><a href="#cb32-199" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-200"><a href="#cb32-200" aria-hidden="true" tabindex="-1"></a>combinations <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb32-201"><a href="#cb32-201" aria-hidden="true" tabindex="-1"></a>  <span class="at">contrast01 =</span> de_01, <span class="at">contrast02 =</span> de_02, <span class="at">contrast03 =</span> de_03,</span>
<span id="cb32-202"><a href="#cb32-202" aria-hidden="true" tabindex="-1"></a>  <span class="at">contrast04 =</span> de_04, <span class="at">contrast05 =</span> de_05, <span class="at">contrast06 =</span> de_06,</span>
<span id="cb32-203"><a href="#cb32-203" aria-hidden="true" tabindex="-1"></a>  <span class="at">contrast07 =</span> de_07, <span class="at">contrast12 =</span> de_12, <span class="at">contrast13 =</span> de_13,</span>
<span id="cb32-204"><a href="#cb32-204" aria-hidden="true" tabindex="-1"></a>  <span class="at">contrast14 =</span> de_14, <span class="at">contrast15 =</span> de_15, <span class="at">contrast16 =</span> de_16,</span>
<span id="cb32-205"><a href="#cb32-205" aria-hidden="true" tabindex="-1"></a>  <span class="at">contrast17 =</span> de_17, <span class="at">contrast23 =</span> de_23, <span class="at">contrast24 =</span> de_24,</span>
<span id="cb32-206"><a href="#cb32-206" aria-hidden="true" tabindex="-1"></a>  <span class="at">contrast25 =</span> de_25, <span class="at">contrast26 =</span> de_26, <span class="at">contrast27 =</span> de_27,</span>
<span id="cb32-207"><a href="#cb32-207" aria-hidden="true" tabindex="-1"></a>  <span class="at">contrast34 =</span> de_34, <span class="at">contrast35 =</span> de_35, <span class="at">contrast36 =</span> de_36,</span>
<span id="cb32-208"><a href="#cb32-208" aria-hidden="true" tabindex="-1"></a>  <span class="at">contrast37 =</span> de_37, <span class="at">contrast45 =</span> de_45, <span class="at">contrast46 =</span> de_46,</span>
<span id="cb32-209"><a href="#cb32-209" aria-hidden="true" tabindex="-1"></a>  <span class="at">contrast47 =</span> de_47, <span class="at">contrast56 =</span> de_56, <span class="at">contrast57 =</span> de_57,</span>
<span id="cb32-210"><a href="#cb32-210" aria-hidden="true" tabindex="-1"></a>  <span class="at">contrast67 =</span> de_67</span>
<span id="cb32-211"><a href="#cb32-211" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">compact</span>()</span>
<span id="cb32-212"><a href="#cb32-212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-213"><a href="#cb32-213" aria-hidden="true" tabindex="-1"></a><span class="co"># Iterate through the 'combinations' list</span></span>
<span id="cb32-214"><a href="#cb32-214" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (contrast_name <span class="cf">in</span> <span class="fu">names</span>(combinations)) {</span>
<span id="cb32-215"><a href="#cb32-215" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create a unique file name for each data frame</span></span>
<span id="cb32-216"><a href="#cb32-216" aria-hidden="true" tabindex="-1"></a>  file_name <span class="ot">&lt;-</span> <span class="fu">paste0</span>(contrast_name, <span class="st">"_result.csv"</span>)</span>
<span id="cb32-217"><a href="#cb32-217" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-218"><a href="#cb32-218" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Save the data frame as a CSV file</span></span>
<span id="cb32-219"><a href="#cb32-219" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_csv</span>(combinations[[contrast_name]], <span class="fu">here</span>(tables_dir, file_name))</span>
<span id="cb32-220"><a href="#cb32-220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-221"><a href="#cb32-221" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Print a message to indicate progress</span></span>
<span id="cb32-222"><a href="#cb32-222" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Saved: "</span>, file_name))</span>
<span id="cb32-223"><a href="#cb32-223" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Saved: contrast01_result.csv"
[1] "Saved: contrast02_result.csv"
[1] "Saved: contrast03_result.csv"
[1] "Saved: contrast04_result.csv"
[1] "Saved: contrast05_result.csv"
[1] "Saved: contrast06_result.csv"
[1] "Saved: contrast07_result.csv"
[1] "Saved: contrast12_result.csv"
[1] "Saved: contrast13_result.csv"
[1] "Saved: contrast14_result.csv"
[1] "Saved: contrast15_result.csv"
[1] "Saved: contrast16_result.csv"
[1] "Saved: contrast17_result.csv"
[1] "Saved: contrast23_result.csv"
[1] "Saved: contrast24_result.csv"
[1] "Saved: contrast25_result.csv"
[1] "Saved: contrast26_result.csv"
[1] "Saved: contrast27_result.csv"
[1] "Saved: contrast34_result.csv"
[1] "Saved: contrast35_result.csv"
[1] "Saved: contrast36_result.csv"
[1] "Saved: contrast37_result.csv"
[1] "Saved: contrast45_result.csv"
[1] "Saved: contrast46_result.csv"
[1] "Saved: contrast47_result.csv"
[1] "Saved: contrast56_result.csv"
[1] "Saved: contrast57_result.csv"
[1] "Saved: contrast67_result.csv"</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (contrast <span class="cf">in</span> <span class="fu">names</span>(combinations)) {</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>  top_upregulated <span class="ot">&lt;-</span> combinations[[contrast]] <span class="sc">%&gt;%</span></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(p_val_adj <span class="sc">&lt;</span> <span class="fl">0.05</span>, avg_log2FC <span class="sc">&gt;</span> <span class="dv">0</span>, gene <span class="sc">%in%</span> gene_int) <span class="sc">%&gt;%</span></span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">desc</span>(avg_log2FC)) <span class="sc">%&gt;%</span></span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">5</span>)</span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>  top_downregulated <span class="ot">&lt;-</span> combinations[[contrast]] <span class="sc">%&gt;%</span></span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(p_val_adj <span class="sc">&lt;</span> <span class="fl">0.05</span>, avg_log2FC <span class="sc">&lt;</span> <span class="dv">0</span>, gene <span class="sc">%in%</span> gene_int) <span class="sc">%&gt;%</span></span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(avg_log2FC) <span class="sc">%&gt;%</span></span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">5</span>)</span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-11"><a href="#cb34-11" aria-hidden="true" tabindex="-1"></a>  top_genes[[contrast]] <span class="ot">&lt;-</span> <span class="fu">c</span>(top_upregulated<span class="sc">$</span>gene, top_downregulated<span class="sc">$</span>gene)</span>
<span id="cb34-12"><a href="#cb34-12" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-layout-align="center">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DefaultAssay</span>(srt) <span class="ot">&lt;-</span> <span class="st">"RNA"</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>cluster.averages <span class="ot">&lt;-</span> <span class="fu">AverageExpression</span>(srt, <span class="at">return.seurat =</span> <span class="cn">TRUE</span>)</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a><span class="fu">DefaultAssay</span>(cluster.averages) <span class="ot">&lt;-</span> <span class="st">"RNA"</span></span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>cluster.averages</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>An object of class Seurat 
21989 features across 8 samples within 1 assay 
Active assay: RNA (21989 features, 0 variable features)</code></pre>
</div>
</div>
<div class="cell" data-layout-align="center">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>plot_pairs <span class="ot">&lt;-</span> <span class="cf">function</span>(highlight_genes, pair_name) {</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>  i <span class="ot">&lt;-</span> pair_name <span class="sc">%&gt;%</span> <span class="fu">str_sub</span>(<span class="dv">9</span>, <span class="dv">9</span>)</span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>  j <span class="ot">&lt;-</span> pair_name <span class="sc">%&gt;%</span> <span class="fu">str_sub</span>(<span class="dv">10</span>, <span class="dv">10</span>)</span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Comparison: "</span>, i, <span class="st">" vs "</span>, j))</span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a>  plot <span class="ot">&lt;-</span> <span class="fu">CellScatter</span>(cluster.averages, <span class="at">cell1 =</span> i, <span class="at">cell2 =</span> j, <span class="at">features =</span> gene_int[gene_int <span class="sc">%in%</span> <span class="fu">rownames</span>(cluster.averages)], <span class="at">highlight =</span> highlight_genes)</span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(plot)</span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb37-8"><a href="#cb37-8" aria-hidden="true" tabindex="-1"></a>plots <span class="ot">&lt;-</span> top_genes <span class="sc">%&gt;%</span> <span class="fu">imap</span>(plot_pairs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Comparison: 0 vs 1"
[1] "Comparison: 0 vs 2"
[1] "Comparison: 0 vs 3"
[1] "Comparison: 0 vs 4"
[1] "Comparison: 0 vs 5"
[1] "Comparison: 0 vs 6"
[1] "Comparison: 0 vs 7"
[1] "Comparison: 1 vs 2"
[1] "Comparison: 1 vs 3"
[1] "Comparison: 1 vs 4"
[1] "Comparison: 1 vs 5"
[1] "Comparison: 1 vs 6"
[1] "Comparison: 1 vs 7"
[1] "Comparison: 2 vs 3"
[1] "Comparison: 2 vs 4"
[1] "Comparison: 2 vs 5"
[1] "Comparison: 2 vs 6"
[1] "Comparison: 2 vs 7"
[1] "Comparison: 3 vs 4"
[1] "Comparison: 3 vs 5"
[1] "Comparison: 3 vs 6"
[1] "Comparison: 3 vs 7"
[1] "Comparison: 4 vs 5"
[1] "Comparison: 4 vs 6"
[1] "Comparison: 4 vs 7"
[1] "Comparison: 5 vs 6"
[1] "Comparison: 5 vs 7"
[1] "Comparison: 6 vs 7"</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>plt_grid <span class="ot">&lt;-</span> cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(<span class="at">plotlist =</span> plots, <span class="at">ncol =</span> <span class="dv">7</span>, <span class="at">labels =</span> <span class="fu">names</span>(top_genes) <span class="sc">%&gt;%</span> <span class="fu">map_chr</span>(str_sub, <span class="dv">9</span>, <span class="dv">10</span>), <span class="at">label_size =</span> <span class="dv">8</span>)</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>plt_grid</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="04-de_test-dendrogram_files/figure-html/de-pairs-plot-1.png" class="img-fluid figure-img" width="8400"></p>
</figure>
</div>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save_plot</span>(<span class="fu">here</span>(plots_dir, <span class="st">"de-pairs-plot.pdf"</span>), plt_grid, <span class="at">base_width =</span> <span class="dv">6</span>, <span class="at">base_height =</span> <span class="dv">7</span>, <span class="at">base_asp =</span> <span class="dv">1</span>, <span class="at">ncol =</span> <span class="dv">7</span>, <span class="at">limitsize =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-layout-align="center">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co"># regsrt &lt;- NormalizeData(regsrt)</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a><span class="co"># regsrt &lt;- FindVariableFeatures(regsrt, selection.method = "vst")</span></span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a><span class="co"># all.genes &lt;- rownames(regsrt)</span></span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a><span class="co"># regsrt &lt;- ScaleData(regsrt, features = all.genes)</span></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>all_regulon_markers_regulons <span class="ot">&lt;-</span> <span class="fu">FindAllMarkers</span>(<span class="at">object =</span> regsrt, <span class="at">verbose =</span> F, <span class="at">test.use =</span> <span class="st">"LR"</span>, <span class="at">assay =</span> <span class="st">"Regulons"</span>, <span class="at">only.pos =</span> T, <span class="at">min.pct =</span> <span class="fl">0.05</span>, <span class="at">logfc.threshold =</span> <span class="fl">0.1</span>)</span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(all_regulon_markers_regulons, <span class="fu">here</span>(data_dir, <span class="fu">sprintf</span>(<span class="st">"resolved_subregions_by_microclusters/%s-all-marker-regulons.csv"</span>, bioproject)))</span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">dim</span>(all_regulon_markers_regulons)[<span class="dv">1</span>] <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a>  all_regulon_markers_regulons <span class="ot">&lt;-</span> all_regulon_markers_regulons <span class="sc">%&gt;%</span></span>
<span id="cb41-10"><a href="#cb41-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span></span>
<span id="cb41-11"><a href="#cb41-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">desc</span>(avg_log2FC)) <span class="sc">%&gt;%</span></span>
<span id="cb41-12"><a href="#cb41-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(cluster)</span>
<span id="cb41-13"><a href="#cb41-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-14"><a href="#cb41-14" aria-hidden="true" tabindex="-1"></a>  top_5r <span class="ot">&lt;-</span> <span class="fu">Extract_Top_Markers</span>(<span class="at">marker_dataframe =</span> all_regulon_markers_regulons, <span class="at">num_genes =</span> <span class="dv">5</span>, <span class="at">group_by =</span> <span class="st">"cluster"</span>, <span class="at">gene_column =</span> <span class="st">"gene"</span>, <span class="at">rank_by =</span> <span class="st">"avg_log2FC"</span>, <span class="at">make_unique =</span> <span class="cn">TRUE</span>, <span class="at">named_vector =</span> <span class="cn">FALSE</span>)</span>
<span id="cb41-15"><a href="#cb41-15" aria-hidden="true" tabindex="-1"></a>  top_15r <span class="ot">&lt;-</span> <span class="fu">Extract_Top_Markers</span>(<span class="at">marker_dataframe =</span> all_regulon_markers_regulons, <span class="at">num_genes =</span> <span class="dv">15</span>, <span class="at">group_by =</span> <span class="st">"cluster"</span>, <span class="at">gene_column =</span> <span class="st">"gene"</span>, <span class="at">rank_by =</span> <span class="st">"avg_log2FC"</span>)</span>
<span id="cb41-16"><a href="#cb41-16" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-layout-align="center">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">Iterate_FeaturePlot_scCustom</span>(<span class="at">reduction =</span> <span class="st">"densmap_xgboost"</span>,</span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">na_cutoff =</span> <span class="cn">NA</span>,</span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">pt.size =</span> <span class="dv">4</span>,</span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">order =</span> <span class="cn">TRUE</span>,</span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">alpha_na_exp =</span> <span class="fl">0.1</span>,</span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">alpha_exp =</span> <span class="fl">0.75</span>, <span class="at">seurat_object =</span> regsrt, <span class="at">gene_list =</span> <span class="fu">rownames</span>(regmtx), <span class="at">single_pdf =</span> <span class="cn">FALSE</span>, <span class="at">file_path =</span> plots_dir, <span class="at">file_type=</span><span class="st">".pdf"</span>, <span class="at">file_name =</span> <span class="fu">sprintf</span>(<span class="st">"%s-regulon.pdf"</span>, bioproject), <span class="at">colors_use =</span> regsrt<span class="sc">@</span>misc<span class="sc">$</span>expr_Colour_Pal</span>
<span id="cb42-7"><a href="#cb42-7" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-layout-align="center">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">Clustered_DotPlot</span>(<span class="at">seurat_object =</span> regsrt, <span class="at">colors_use_exp =</span> <span class="fu">viridis</span>(<span class="at">n =</span> <span class="dv">30</span>, <span class="at">alpha =</span> .<span class="dv">75</span>, <span class="at">direction =</span> <span class="sc">-</span><span class="dv">1</span>, <span class="at">option =</span> <span class="st">"G"</span>), <span class="at">features =</span> <span class="fu">rownames</span>(regmtx), <span class="at">k =</span> <span class="dv">14</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="04-de_test-dendrogram_files/figure-html/plt-dotplot-dendrogram-regulons-1.png" class="img-fluid figure-img" width="5400"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[[1]]</code></pre>
</div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="04-de_test-dendrogram_files/figure-html/plt-dotplot-dendrogram-regulons-2.png" class="img-fluid figure-img" width="5400"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
[[2]]</code></pre>
</div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="04-de_test-dendrogram_files/figure-html/plt-dotplot-dendrogram-regulons-3.png" class="img-fluid figure-img" width="5400"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="session-information" class="level2">
<h2 class="anchored" data-anchor-id="session-information">Session information</h2>
<div class="cell" data-layout-align="center">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>sI <span class="ot">&lt;-</span> sessioninfo<span class="sc">::</span><span class="fu">session_info</span>()</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>sI<span class="sc">$</span>loadedOnly <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(sI, <span class="at">locale =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>─ Session info ───────────────────────────────────────────────────────────────
 setting  value
 version  R version 4.2.2 (2022-10-31)
 os       Ubuntu 22.04.2 LTS
 system   x86_64, linux-gnu
 ui       X11
 language en_US:en
 collate  en_US.UTF-8
 ctype    en_US.UTF-8
 tz       Etc/UTC
 date     2023-07-07
 pandoc   2.19.2 @ /opt/python/3.8.8/bin/ (via rmarkdown)

─ Packages ───────────────────────────────────────────────────────────────────
 package              * version     date (UTC) lib source
 abind                  1.4-5       2016-07-21 [1] RSPM (R 4.2.0)
 anndata              * 0.7.5.6     2023-03-17 [1] RSPM (R 4.2.0)
 assertthat             0.2.1       2019-03-21 [1] RSPM (R 4.2.0)
 base64enc              0.1-3       2015-07-28 [1] RSPM (R 4.2.0)
 bayestestR             0.13.1      2023-04-30 [1] https://easystats.r-universe.dev (R 4.2.2)
 beeswarm               0.4.0       2021-06-01 [1] RSPM (R 4.2.0)
 Biobase                2.58.0      2022-11-01 [1] RSPM (R 4.2.2)
 BiocGenerics           0.44.0      2022-11-01 [1] RSPM (R 4.2.2)
 BiocManager            1.30.20     2023-02-24 [1] RSPM (R 4.2.0)
 bit                    4.0.5       2022-11-15 [1] RSPM (R 4.2.0)
 bit64                  4.0.5       2020-08-30 [1] RSPM (R 4.2.0)
 bitops                 1.0-7       2021-04-24 [1] RSPM (R 4.2.0)
 Cairo                  1.6-0       2022-07-05 [1] RSPM (R 4.2.2)
 circlize               0.4.15      2022-05-10 [1] RSPM (R 4.2.0)
 cli                    3.6.1       2023-03-23 [1] RSPM (R 4.2.0)
 clue                   0.3-64      2023-01-31 [1] RSPM (R 4.2.0)
 cluster                2.1.4       2022-08-22 [1] RSPM (R 4.2.0)
 coda                   0.19-4      2020-09-30 [1] RSPM (R 4.2.0)
 codetools              0.2-19      2023-02-01 [1] RSPM (R 4.2.0)
 colorspace             2.1-0       2023-01-23 [1] RSPM (R 4.2.0)
 ComplexHeatmap         2.14.0      2022-11-01 [1] RSPM (R 4.2.2)
 correlation            0.8.4       2023-04-30 [1] Github (easystats/correlation@3d5cd1e)
 cowplot              * 1.1.1       2020-12-30 [1] RSPM (R 4.2.0)
 crayon                 1.5.2       2022-09-29 [1] RSPM (R 4.2.0)
 data.table             1.14.8      2023-02-17 [1] RSPM (R 4.2.0)
 datawizard             0.7.1       2023-04-03 [1] RSPM (R 4.2.0)
 DelayedArray           0.24.0      2022-11-01 [1] RSPM (R 4.2.2)
 deldir                 1.0-6       2021-10-23 [1] RSPM (R 4.2.0)
 digest                 0.6.31      2022-12-11 [1] RSPM (R 4.2.0)
 doParallel             1.0.17      2022-02-07 [1] RSPM (R 4.2.0)
 dplyr                * 1.1.2       2023-04-20 [1] RSPM (R 4.2.2)
 ellipsis               0.3.2       2021-04-29 [1] RSPM (R 4.2.0)
 emmeans                1.8.5       2023-03-08 [1] RSPM (R 4.2.0)
 estimability           1.4.1       2022-08-05 [1] RSPM (R 4.2.0)
 evaluate               0.20        2023-01-17 [1] RSPM (R 4.2.0)
 fansi                  1.0.4       2023-01-22 [1] RSPM (R 4.2.0)
 farver                 2.1.1       2022-07-06 [1] RSPM (R 4.2.0)
 fastmap                1.1.1       2023-02-24 [1] RSPM (R 4.2.0)
 fitdistrplus           1.1-11      2023-04-25 [1] RSPM (R 4.2.2)
 forcats              * 1.0.0       2023-01-29 [1] RSPM (R 4.2.0)
 foreach                1.5.2       2022-02-02 [1] RSPM (R 4.2.0)
 future               * 1.32.0      2023-03-07 [1] RSPM (R 4.2.0)
 future.apply           1.10.0      2022-11-05 [1] RSPM (R 4.2.0)
 generics               0.1.3       2022-07-05 [1] RSPM (R 4.2.0)
 GenomeInfoDb           1.34.9      2023-02-02 [1] RSPM (R 4.2.2)
 GenomeInfoDbData       1.2.9       2023-04-30 [1] RSPM (R 4.2.2)
 GenomicRanges          1.50.2      2022-12-16 [1] RSPM (R 4.2.2)
 GetoptLong             1.0.5       2020-12-15 [1] RSPM (R 4.2.0)
 ggbeeswarm             0.7.2       2023-04-30 [1] Github (eclarke/ggbeeswarm@3cf58a9)
 ggmin                  0.0.0.9000  2023-04-30 [1] Github (sjessa/ggmin@8ada274)
 ggplot2              * 3.4.2       2023-04-03 [1] RSPM (R 4.2.0)
 ggprism                1.0.4       2023-04-30 [1] Github (csdaw/ggprism@0e411f4)
 ggrastr                1.0.1       2023-04-30 [1] Github (VPetukhov/ggrastr@7aed9af)
 ggrepel                0.9.2.9999  2023-04-30 [1] Github (slowkow/ggrepel@fe3b5c3)
 ggridges               0.5.4       2022-09-26 [1] RSPM (R 4.2.0)
 ggstatsplot          * 0.11.1.9000 2023-04-30 [1] Github (IndrajeetPatil/ggstatsplot@befe812)
 GlobalOptions          0.1.2       2020-06-10 [1] RSPM (R 4.2.0)
 globals                0.16.2      2022-11-21 [1] RSPM (R 4.2.0)
 glue                   1.6.2       2022-02-24 [1] RSPM (R 4.2.0)
 goftest                1.2-3       2021-10-07 [1] RSPM (R 4.2.0)
 gridExtra              2.3         2017-09-09 [1] RSPM (R 4.2.0)
 gtable                 0.3.3       2023-03-21 [1] RSPM (R 4.2.0)
 hdf5r                  1.3.8       2023-01-21 [1] RSPM (R 4.2.2)
 here                 * 1.0.1       2020-12-13 [1] RSPM (R 4.2.0)
 hms                    1.1.3       2023-03-21 [1] RSPM (R 4.2.0)
 htmltools              0.5.5       2023-03-23 [1] RSPM (R 4.2.0)
 htmlwidgets            1.6.2       2023-03-17 [1] RSPM (R 4.2.0)
 httpuv                 1.6.9       2023-02-14 [1] RSPM (R 4.2.0)
 httr                   1.4.5       2023-02-24 [1] RSPM (R 4.2.0)
 ica                    1.0-3       2022-07-08 [1] RSPM (R 4.2.0)
 igraph                 1.4.1       2023-02-24 [1] RSPM (R 4.2.0)
 insight                0.19.1      2023-03-18 [1] RSPM (R 4.2.0)
 IRanges                2.32.0      2022-11-01 [1] RSPM (R 4.2.2)
 irlba                  2.3.5.1     2022-10-03 [1] RSPM (R 4.2.0)
 iterators              1.0.14      2022-02-05 [1] RSPM (R 4.2.0)
 janitor                2.2.0.9000  2023-04-30 [1] Github (sfirke/janitor@d64c8bb)
 jsonlite               1.8.4       2022-12-06 [1] RSPM (R 4.2.0)
 KernSmooth             2.23-20     2021-05-03 [1] RSPM (R 4.2.0)
 knitr                  1.42        2023-01-25 [1] RSPM (R 4.2.0)
 labeling               0.4.2       2020-10-20 [1] RSPM (R 4.2.0)
 later                  1.3.0       2021-08-18 [1] RSPM (R 4.2.0)
 lattice                0.21-8      2023-04-05 [1] RSPM (R 4.2.0)
 lazyeval               0.2.2       2019-03-15 [1] RSPM (R 4.2.0)
 leiden                 0.4.3       2022-09-10 [1] RSPM (R 4.2.0)
 lifecycle              1.0.3       2022-10-07 [1] RSPM (R 4.2.0)
 listenv                0.9.0       2022-12-16 [1] RSPM (R 4.2.0)
 lmtest                 0.9-40      2022-03-21 [1] RSPM (R 4.2.0)
 lubridate            * 1.9.2       2023-02-10 [1] RSPM (R 4.2.0)
 magick                 2.7.4       2023-03-09 [1] RSPM (R 4.2.0)
 magrittr             * 2.0.3       2022-03-30 [1] RSPM (R 4.2.0)
 MASS                   7.3-58.1    2022-08-03 [1] CRAN (R 4.2.2)
 MAST                   1.24.1      2023-01-23 [1] RSPM (R 4.2.2)
 Matrix                 1.5-4       2023-04-04 [1] RSPM (R 4.2.0)
 MatrixGenerics         1.10.0      2022-11-01 [1] RSPM (R 4.2.2)
 matrixStats            0.63.0      2022-11-18 [1] RSPM (R 4.2.0)
 mime                   0.12        2021-09-28 [1] RSPM (R 4.2.0)
 miniUI                 0.1.1.1     2018-05-18 [1] RSPM (R 4.2.0)
 multcomp               1.4-23      2023-03-09 [1] RSPM (R 4.2.0)
 munsell                0.5.0       2018-06-12 [1] RSPM (R 4.2.0)
 mvtnorm                1.1-3       2021-10-08 [1] RSPM (R 4.2.0)
 nlme                   3.1-162     2023-01-31 [1] RSPM (R 4.2.0)
 paletteer              1.5.0       2022-10-19 [1] RSPM (R 4.2.0)
 parallelly             1.35.0      2023-03-23 [1] RSPM (R 4.2.0)
 parameters             0.20.3      2023-04-05 [1] RSPM (R 4.2.0)
 patchwork            * 1.1.2.9000  2023-04-30 [1] Github (thomasp85/patchwork@c14c960)
 pbapply                1.7-0       2023-01-13 [1] RSPM (R 4.2.0)
 pillar                 1.9.0       2023-03-22 [1] RSPM (R 4.2.0)
 pkgconfig              2.0.3       2019-09-22 [1] RSPM (R 4.2.0)
 plotly                 4.10.1      2022-11-07 [1] RSPM (R 4.2.0)
 plyr                   1.8.8       2022-11-11 [1] RSPM (R 4.2.0)
 png                    0.1-8       2022-11-29 [1] RSPM (R 4.2.0)
 polyclip               1.10-4      2022-10-20 [1] RSPM (R 4.2.0)
 prettyunits            1.1.1       2020-01-24 [1] RSPM (R 4.2.0)
 prismatic              1.1.1       2022-08-15 [1] RSPM (R 4.2.0)
 progress               1.2.2       2019-05-16 [1] RSPM (R 4.2.0)
 progressr              0.13.0      2023-01-10 [1] RSPM (R 4.2.0)
 promises               1.2.0.1     2021-02-11 [1] RSPM (R 4.2.0)
 purrr                * 1.0.1       2023-01-10 [1] RSPM (R 4.2.0)
 R.methodsS3            1.8.2       2022-06-13 [1] RSPM (R 4.2.0)
 R.oo                   1.25.0      2022-06-12 [1] RSPM (R 4.2.0)
 R.utils                2.12.2      2022-11-11 [1] RSPM (R 4.2.0)
 R6                     2.5.1       2021-08-19 [1] RSPM (R 4.2.0)
 ragg                   1.2.5       2023-01-12 [1] RSPM (R 4.2.0)
 RANN                   2.6.1       2019-01-08 [1] RSPM (R 4.2.0)
 RColorBrewer         * 1.1-3       2022-04-03 [1] RSPM (R 4.2.0)
 Rcpp                   1.0.10      2023-01-22 [1] RSPM (R 4.2.0)
 RcppAnnoy              0.0.20      2022-10-27 [1] RSPM (R 4.2.0)
 RCurl                  1.98-1.12   2023-03-27 [1] RSPM (R 4.2.0)
 readr                * 2.1.4       2023-02-10 [1] RSPM (R 4.2.0)
 rematch2               2.1.2       2020-05-01 [1] RSPM (R 4.2.0)
 remotes                2.4.2       2021-11-30 [1] RSPM (R 4.2.0)
 repr                   1.1.6       2023-01-26 [1] RSPM (R 4.2.0)
 reshape2               1.4.4       2020-04-09 [1] RSPM (R 4.2.0)
 reticulate           * 1.28-9000   2023-04-30 [1] Github (rstudio/reticulate@442c49f)
 rjson                  0.2.21      2022-01-09 [1] RSPM (R 4.2.0)
 rlang                  1.1.0       2023-03-14 [1] RSPM (R 4.2.0)
 rmarkdown              2.21        2023-03-26 [1] RSPM (R 4.2.0)
 ROCR                   1.0-11      2020-05-02 [1] RSPM (R 4.2.0)
 rprojroot              2.0.3       2022-04-02 [1] RSPM (R 4.2.0)
 rsvd                   1.0.5       2021-04-16 [1] RSPM (R 4.2.0)
 Rtsne                  0.16        2022-04-17 [1] RSPM (R 4.2.0)
 S4Vectors              0.36.2      2023-02-26 [1] RSPM (R 4.2.2)
 sandwich               3.0-2       2022-06-15 [1] RSPM (R 4.2.0)
 scales                 1.2.1       2022-08-20 [1] RSPM (R 4.2.0)
 scattermore            0.8         2022-02-14 [1] RSPM (R 4.2.0)
 scCustomize          * 1.1.1       2023-04-30 [1] Github (samuel-marsh/scCustomize@d08268d)
 sceasy               * 0.0.7       2023-04-30 [1] Github (cellgeni/sceasy@0cfc0e3)
 sctransform            0.3.5       2022-09-21 [1] RSPM (R 4.2.0)
 sessioninfo            1.2.2       2021-12-06 [1] RSPM (R 4.2.0)
 Seurat               * 4.3.0       2022-11-18 [1] RSPM (R 4.2.2)
 SeuratDisk           * 0.0.0.9020  2023-04-30 [1] Github (mojaveazure/seurat-disk@9b89970)
 SeuratObject         * 4.1.3       2022-11-07 [1] RSPM (R 4.2.0)
 SeuratWrappers       * 0.3.1       2023-04-30 [1] Github (satijalab/seurat-wrappers@d28512f)
 shape                  1.4.6       2021-05-19 [1] RSPM (R 4.2.0)
 shiny                  1.7.4       2022-12-15 [1] RSPM (R 4.2.0)
 SingleCellExperiment   1.20.1      2023-03-17 [1] RSPM (R 4.2.2)
 skimr                * 2.1.5       2023-04-30 [1] Github (ropensci/skimr@d5126aa)
 snakecase              0.11.0      2019-05-25 [1] RSPM (R 4.2.0)
 sp                     1.6-0       2023-01-19 [1] RSPM (R 4.2.0)
 spatstat.data          3.0-1       2023-03-12 [1] RSPM (R 4.2.0)
 spatstat.explore       3.1-0       2023-03-14 [1] RSPM (R 4.2.0)
 spatstat.geom          3.1-0       2023-03-12 [1] RSPM (R 4.2.0)
 spatstat.random        3.1-4       2023-03-13 [1] RSPM (R 4.2.0)
 spatstat.sparse        3.0-1       2023-03-12 [1] RSPM (R 4.2.0)
 spatstat.utils         3.0-2       2023-03-11 [1] RSPM (R 4.2.0)
 statsExpressions       1.5.0       2023-02-19 [1] RSPM (R 4.2.0)
 stringi                1.7.12      2023-01-11 [1] RSPM (R 4.2.0)
 stringr              * 1.5.0       2022-12-02 [1] RSPM (R 4.2.0)
 SummarizedExperiment   1.28.0      2022-11-01 [1] RSPM (R 4.2.2)
 survival               3.5-5       2023-03-12 [1] RSPM (R 4.2.0)
 systemfonts            1.0.4       2022-02-11 [1] RSPM (R 4.2.0)
 tensor                 1.5         2012-05-05 [1] RSPM (R 4.2.0)
 textshaping            0.3.6       2021-10-13 [1] RSPM (R 4.2.0)
 TH.data                1.1-1       2022-04-26 [1] RSPM (R 4.2.0)
 tibble               * 3.2.1       2023-03-20 [1] RSPM (R 4.2.0)
 tidyr                * 1.3.0       2023-01-24 [1] RSPM (R 4.2.0)
 tidyselect             1.2.0       2022-10-10 [1] RSPM (R 4.2.0)
 tidyverse            * 2.0.0.9000  2023-04-30 [1] Github (tidyverse/tidyverse@8ec2e1f)
 timechange             0.2.0       2023-01-11 [1] RSPM (R 4.2.0)
 tzdb                   0.3.0       2022-03-28 [1] RSPM (R 4.2.0)
 utf8                   1.2.3       2023-01-31 [1] RSPM (R 4.2.0)
 uwot                   0.1.14      2022-08-22 [1] RSPM (R 4.2.0)
 vctrs                  0.6.2       2023-04-19 [1] RSPM (R 4.2.2)
 vipor                  0.4.5       2017-03-22 [1] RSPM (R 4.2.0)
 viridis              * 0.6.2       2021-10-13 [1] RSPM (R 4.2.0)
 viridisLite          * 0.4.1       2022-08-22 [1] RSPM (R 4.2.0)
 vroom                  1.6.1       2023-01-22 [1] RSPM (R 4.2.0)
 withr                  2.5.0       2022-03-03 [1] RSPM (R 4.2.0)
 xfun                   0.39        2023-04-20 [1] RSPM (R 4.2.2)
 xtable                 1.8-4       2019-04-21 [1] RSPM (R 4.2.0)
 XVector                0.38.0      2022-11-01 [1] RSPM (R 4.2.2)
 yaml                   2.3.7       2023-01-23 [1] RSPM (R 4.2.0)
 zeallot                0.1.0       2018-01-28 [1] RSPM (R 4.2.0)
 zlibbioc               1.44.0      2022-11-01 [1] RSPM (R 4.2.2)
 zoo                    1.8-12      2023-04-13 [1] RSPM (R 4.2.2)

 [1] /opt/R/4.2.2/lib/R/library

─ Python configuration ───────────────────────────────────────────────────────
 python:         /opt/python/3.8.8/bin/python
 libpython:      /opt/python/3.8.8/lib/libpython3.8.so
 pythonhome:     /opt/python/3.8.8:/opt/python/3.8.8
 version:        3.8.8 | packaged by conda-forge | (default, Feb 20 2021, 16:22:27)  [GCC 9.3.0]
 numpy:          /opt/python/3.8.8/lib/python3.8/site-packages/numpy
 numpy_version:  1.23.5
 scanpy:         /opt/python/3.8.8/lib/python3.8/site-packages/scanpy
 
 NOTE: Python version was forced by RETICULATE_PYTHON

──────────────────────────────────────────────────────────────────────────────</code></pre>
</div>
</div>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>